### PR TITLE
Add some const fixes and use C++11 features in cli.{hpp,cpp}

### PR DESCRIPTION
**Pull request by dreamer-dead**
Hi!
Could you take a look at this request please?

### CODE REVIEW

File: cli.cpp [hunk 1]
```diff
@@ -15,33 +16,58 @@ using std::string;
...
+    std::vector<char *> cargs;
+    std::vector<std::string> mutableArgs(args);
+    cargs.push_back(program_invocation_name);
+    for (auto& arg : mutableArgs)
+        cargs.push_back(&arg[0]);
+
+    cmd->SetDieOnSignal(false);
+
+    CurrentCmd = cmd.get();
+    int ret = cmd->Execute(cargs.size() - 1, (&cargs[0]) + 1);
```
Comment (dreamer-dead) [inline cli.cpp]: Is ICmd::Execute really need to have a non-const args?
It it possible to pass a `const char*[]` ?
Copy of args - mutableArgs can be removed in that case.

Comment (rgushchin) [inline cli.cpp]: There is some bad magic about passing and not-passing argv[0] into command arguments. It seems to me, that we can simplify the code even more radically.

Comment (dreamer-dead) [inline cli.cpp]: Yes, it was strange to me about adding and skipping argv[0].
Could you explain what should be done with arguments of a command and program name?
And then I will try to simplify it.
In this PR I've tried to not change existing behavor of the code.

**Comment by rgushchin** [comment]
Command name is used in general to select which command should be executed. True argv[0] is never used (or at least I can't find where it's used).

I think, we can do the following to simplify everything:
1) move statically defined args of GetOpt() call to ICmd ancestors (as private data field),
2) move GetOpt() call to ICmd::TryExec() method,
3) remove all this ugly "argv-1/argc+1" stuff.

**Comment by dreamer-dead** [comment]
I've rebased on latest master.
Here is my attempt to get rid of C-style argc/argv command arguments handling.
It's done not exactly in the way you mentioned before, and I'm ready to discuss it =)

But now we don't have C-style arrays-of-arrays of chars.
And static map with commands is gone.

Unfortunately, my changes lead to a huge diff in portoctl.cpp and it is hard to review (
Please take a look and say what do you think about this code.

File: src/cli.cpp [hunk 2]
```diff
@@ -187,64 +126,146 @@ void THelpCmd::Usage() {
...
+
+    CurrentCmd = cmd.get();
+
+    TCommandEnviroment childEnv(env, args);
+    int ret = cmd->Execute(&childEnv);
+    CurrentCmd = prevCmd;
 
-    cmd->Execute(command ? 1 : 0, argv);
+    if (GotSignal())
+        return InterruptedSignal;
```
Comment (dreamer-dead) [inline src/cli.cpp]: During this refactoring I got some questions.
Could you give me some advices?
For example here we return just a `InterruptedSignal` value. But there is code line in TCommandHandler::TryExec where we return `-cmd->InterruptedSignal;` 
Is that OK?

File: src/cli.cpp [hunk 3]
```diff
@@ -313,9 +360,14 @@ int GetOpt(int argc, char *argv[], const std::vector<Option> &opts) {
 
     optind = 1;
     int opt;
-    while ((opt = getopt(argc + 1, argv - 1, optstring.c_str())) != -1) {
+    vector<string> mutableBuffer = Arguments;
+    vector<char*> rawArgs;
+    rawArgs.reserve(mutableBuffer.size());
+    for (auto & arg : mutableBuffer)
+        rawArgs.push_back(&arg[0]);
+    while ((opt = getopt(rawArgs.size(), &rawArgs[0], optstring.c_str())) != -1) {
```
Comment (dreamer-dead) [inline src/cli.cpp]: Here I changed old code behavior and skip previous (-1) command argument.
As I can see, usualy it was just a program name (was added to argument list in ICmd::RunCmd) and getopt will always skip it.
Am I right?

File: src/portoctl.cpp [hunk 4]
```diff
@@ -376,25 +378,25 @@ class TSetPropertyCmd : public ICmd {
...
-class TGetDataCmd : public ICmd {
+class TGetDataCmd final : public ICmd {
 public:
     TGetDataCmd(TPortoAPI *api) : ICmd(api, "dget", 2, "[-k] <container> <data> [data...]", "get raw container data") {}
 
-    int Execute(int argc, char *argv[]) {
+    int Execute(TCommandEnviroment *env) final override {
         bool printKey = false;
-        int start = GetOpt(argc, argv, {
+        const auto &args = env->GetOpts({
```
Comment (dreamer-dead) [inline src/portoctl.cpp]: Unfortunately, I was unable to move these static defined options to class members (
It uses closures with local vars and class-scope vars and such change will lead to massive commands rewrite.

File: src/portoctl.cpp [hunk 5]
```diff
@@ -688,15 +693,15 @@ class TEnterCmd : public ICmd {
...
             { 'C', false, [&](const char *arg) { enterCgroups = false; } },
         });
 
-        string cmd = "";
-        for (int i = start + 1; i < argc; i++) {
-            cmd += argv[i];
+        string cmd;
+        for (size_t i = 1; i < args.size(); ++i) {
```
Comment (dreamer-dead) [inline src/portoctl.cpp]: All these changes should be reviewed very carefully.
Are there tests for portoctl functionality ?
All of portoctl command are classes local to portoctl.cpp file and cannot be tested with unit-tests.
Maybe we can have some kind of script to run portoctl comands and check some expects?

**Comment by rgushchin** [comment]
No, there are no dedicated tests. Portoclt utility is used in a number of existing tests, although I don't think, they cover their functionality somehow.
Yes, it's a good idea to write portoctl tests.

**Comment by rgushchin** [comment]
I'll try to find some time to review your changes today or at Monday.
Thank you!

**Comment by rgushchin** [comment]
Merged.
Thank you!

**Pull request merged**